Cpp_test/lib/file-handler.cpp

```diff
@@ -0,0 +1,70 @@
+#include "file-handler.h"
+#include <iostream>
+
+
+FileHandler::~FileHandler()
+{
+	if (fs.is_open())
+		fs.close();
+}
+
+
+void FileHandler::open_file()
+{
+	if (this->mode == FileMode::READ)
```

Comment:
I think personally I would either move this mode to a parameter in this function, or do this open_file functionality in the constructor already. It's not immediately clear where this mode comes from, so it could be a bit clearer if it was a parameter or this opening were done in constructor. I believe C++ advocates this type of resource allocation anyway (RAII), where the constructor already allocates resources.

Reply:
I will change this->mode to be a specific typo rule to see variables within a class so it will be clearer